lib/utils.js

```diff
@@ -19,6 +19,35 @@ function shallowCopy(src, dest) {
   return dest;
 }
 
+function take(source, fromRight) {
+  return fromRight ? source.pop() : source.shift();
+}
+
+/**
+ * Takes a list of arguments and maps them to a list of names
+ * in an alternating either a left-to-right, or right-to-left direction.
+ */
+function resolveArgs(argumentsObject, nameMap, rightToLeft) {
+  var result = {};
+
+  var takeFromRight = rightToLeft;
+  var args = Array.prototype.slice.call(argumentsObject);
+
+  while (nameMap.length) {
+    var value = null;
+    var name = take(nameMap, takeFromRight);
+
+    if (args.length > 0) {
+      value = take(args, takeFromRight);
+      takeFromRight = !takeFromRight;
```

Comment:
Timothy asked the same question. You can see my answer here:

https://github.com/stormpath/stormpath-sdk-node/commit/4e24e1e89b4e1900d89fbe6c54cc31e2f8261a66#diff-50e3aa130a4f97a42ee2cf111c7b1d9dR42